### PR TITLE
Set global object's prototype

### PR DESCRIPTION
**Pull request by galpeter**
The global object should have the same prototype
as a simple object.

JerryScript-DCO-1.0-Signed-off-by: Peter Gal pgal.u-szeged@partner.samsung.com

### CODE REVIEW

**Comment by galpeter** [comment]
Related test262 test: https://github.com/tc39/test262/blob/es5-tests/test/suite/ch15/15.2/15.2.3/15.2.3.14/15.2.3.14-6-6.js

File: jerry-core/ecma/builtin-objects/ecma-builtins.inc.h [hunk 1]
```diff
@@ -301,7 +301,7 @@ BUILTIN (ECMA_BUILTIN_ID_COMPACT_PROFILE_ERROR,
 /* The Global object (15.1) */
 BUILTIN (ECMA_BUILTIN_ID_GLOBAL,
          ECMA_OBJECT_TYPE_GENERAL,
-         ECMA_BUILTIN_ID__COUNT /* no prototype */,
+         ECMA_BUILTIN_ID_OBJECT_PROTOTYPE,
```
Comment (ruben-ayrapetyan) [inline jerry-core/ecma/builtin-objects/ecma-builtins.inc.h]: According to ECMA-262 v5, the value is implementation-dependent.
Could you, please, add comment about this?

Are there some test cases that depend on the `this.hasOwnProperty` value? Why do we need the change?

File: tests/jerry/regression-test-issue-122.js [hunk 2]
```diff
@@ -15,10 +15,11 @@
 
 try
 {
-  print(this, []);
+  print({toString: function() { throw new TypeError("foo"); }}, []);
```
Comment (ruben-ayrapetyan) [inline tests/jerry/regression-test-issue-122.js]: Seems that the change is not related to the patch.

File: jerry-core/ecma/builtin-objects/ecma-builtins.inc.h (hunk 1 above)
Comment (galpeter) [inline jerry-core/ecma/builtin-objects/ecma-builtins.inc.h]: Ok, I'll add a comment.

Please see the https://github.com/tc39/test262/blob/es5-tests/test/suite/ch15/15.2/15.2.3/15.2.3.14/15.2.3.14-6-6.js testcase's 17 line, where it calls the 'hasOwnProperty' on the global object.

Comment (ruben-ayrapetyan) [inline jerry-core/ecma/builtin-objects/ecma-builtins.inc.h]: Ok. Thank you.

File: tests/jerry/regression-test-issue-122.js (hunk 2 above)
Comment (galpeter) [inline tests/jerry/regression-test-issue-122.js]: It is related, a with this PR the global object can be printed out and won't cause an Error. Due to the fact that the prototype is now set, and thus the toString method will be found when converting the object to string.

Comment (ruben-ayrapetyan) [inline tests/jerry/regression-test-issue-122.js]: I see.

Maybe, now we could use the initial version of regression test case from #122:
`print(this, []);`

**Comment by ruben-ayrapetyan** [comment]
Looks good to me, after adding comment.

File: tests/jerry/regression-test-issue-122.js (hunk 2 above)
Comment (galpeter) [inline tests/jerry/regression-test-issue-122.js]: You mean replace the whole `regression-test-issue-122.js` file with the simple print? But then we will not have a test case which creates an error for the `print` call.

Comment (ruben-ayrapetyan) [inline tests/jerry/regression-test-issue-122.js]: I see. Yes, it would be incorrect to replace with simple print.

The change looks good to me.

**Comment by egavrin** [comment]
`make push`

**Comment by LaszloLango** [comment]
+1 lgtm
